### PR TITLE
Create class

### PR DESCRIPTION
**Pull request by HenaryKhuraijam**
#### Created class Person with the following:
- nstance vars: @id, @name, and @age.
- Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- Getters for @id, @name, and @age.
- Setters for @name and @age.
- Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- Public method can_use_services? returns true if person is of age or if they have permission from parents.

#### Create class Student with the following:
- Inherits from Person.
- Constructor extends parent's constructor by adding @classroom and a parameter for it.
- Method play_hooky that returns "¯\(ツ)/¯".

#### Create class Teacher with the following:
- Inherits from Person.
- Constructor extends parent's constructor by adding @specialization and a parameter for it.
- Override can_use_services? so it always returns true.

### CODE REVIEW

**Review by zeff96 — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW 🛑
Hi @HenaryKhuraijam , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Hey there! 🌟 It appears there's a slight hiccup in the current project submission. 🚧 Please take note that for this milestone, we're aligning with the GitFlow workflow. 🌿 This involves closing the existing pull request without merging and initiating a fresh pull request. 🔄 The new pull request should encompass all the introduced changes in this milestone and be initiated from a feature branch to the development branch. 🚀 Thank you for your attention to this matter! 🙌

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by zeff96** [comment]
@HenaryKhuraijam, Kindly note of the above invalidation reason and make the requested changes for the workflow before submitting for review. use GitFlow as outline in the project [guideline](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md)

File: README.md [hunk 1]
```diff
@@ -1 +1,120 @@
-# oop-school-library
\ No newline at end of file
+
+<!-- TABLE OF CONTENTS -->
+
+# 📗 Table of Contents
+
+- [📗 Table of Contents](#-table-of-contents)
+- [📖 \[decode-morse-code\] ](#-decode-morse-code-)
```
Comment (zeff96) [inline README.md]: - [ ] Greetings! 📚 Just a quick heads-up: this project revolves around an Object-Oriented Programming (OOP) School Library. 🏫✨ Could you please tweak the header in the table of contents to align with this theme? 🔄 Let's ensure the documentation reflects the essence of our OOP-focused endeavor. 🌐 Thanks a bunch! 🙌

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️
Hi @HenaryKhuraijam , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: Correct GitFlow



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (HenaryKhuraijam) [reply README.md]: Done

**Review by KDlamini — APPROVED** [review]
### Hi @HenaryKhuraijam  🙋‍♀️,  

Excellent work! 💯 
You have done a great job adding the Person, Student and Teacher classes for the OOP school library. 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 

✔️  Professional PR title and description
✔️  Well documented and professional README
✔️  There's an individual `.rb` file for each class (`Person`, `Student`, and `Teacher` )
✔️  All class methods and variables are implemented correctly

# Status: Approved ✔️ 

Will_Smith_dance

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**